builder/amazon/common/step_run_source_instance.go

```diff
@@ -114,6 +117,7 @@ func (s *StepRunSourceInstance) Run(ctx context.Context, state multistep.StateBa
 		UserData:            &userData,
 		MaxCount:            aws.Int64(1),
 		MinCount:            aws.Int64(1),
+		MetadataOptions:     &ec2.InstanceMetadataOptionsRequest{HttpEndpoint: &s.HttpEndpoint, HttpTokens: &s.HttpTokens, HttpPutResponseHopLimit: &s.HttpPutResponseHopLimit},
```

Comment:
I think some validation is necessary before this. We need to make sure `HttpEndpoint` is `enabled` before setting the other two variables. 
ref: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html

I suggest you look at the terraform provider logic to implement something similar 
https://github.com/hashicorp/terraform-provider-aws/blob/main/aws/resource_aws_instance.go#L1364

Reply:
I've done some validation for this, it required me to do some sane default setting to make the validation work as expected without flopping back to weird values. Let me know if my implementation is ok.